### PR TITLE
soc: riscv: riscv-privilege: telink: Added Telink b92 soc PMP support

### PR DESCRIPTION
**Pull request by mishadesh**
Added the generic RISCV PMP support for B92 SoC series.

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
Can we create Telink config option for enabling memory protection feature?
Disable it for default (for now) and selects all features you provides.
In tht case it's safe entry point for memory protection implementation.

**Comment by mishadesh** [comment]
> Can we create Telink config option for enabling memory protection feature? Disable it for default (for now) and selects all features you provides. In tht case it's safe entry point for memory protection implementation.

Yes, we can. But the purpose of it is not clear. This is just a description of CPU architecture. So we are not enabling the implementation of user space or memory protection.
Also the ROM size and RAM size is not changing in case if this option is enabled. 

The SiFive vendor just adds this option to his own core description.

So I think for development process we may go the way you suggested, but after we finish, we need to prepare our code in common way.

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 1]
```diff
@@ -16,6 +16,11 @@ config B9xCPU_RISCV32
 
 endchoice
 
+config TELINK_B92_PMP
+	bool "Support Hardware Memory Protection"
+	default n
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc]: By default option is no. Seems like this line is not required.

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 2]
```diff
@@ -52,6 +57,16 @@ config SOC_RISCV_TELINK_B92
 
 endchoice
 
+config PMP_POWER_OF_TWO_ALIGNMENT
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc]: This option is no by default?

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 3]
```diff
@@ -52,6 +57,16 @@ config SOC_RISCV_TELINK_B92
...
+	bool
+	depends on RISCV_PMP
+	default n
+
+config PMP_SLOTS
+	int
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc]: We are changing just value.

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 4]
```diff
@@ -52,6 +56,13 @@ config SOC_RISCV_TELINK_B92
 
 endchoice
 
+config PMP_POWER_OF_TWO_ALIGNMENT
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc]: Why do we still need this line? Now seems it changes nothing.

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 5]
```diff
@@ -52,6 +56,10 @@ config SOC_RISCV_TELINK_B92
 
 endchoice
 
+config PMP_SLOTS
+	depends on RISCV_PMP
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc]:
```
if RISCV_PMP

config PMP_SLOTS
	int "Number of PMP slots"
	default 8
	help
	  This is the number of PMP entries implemented by the hardware.
	  Typical values are 8 or 16.
```
The dependency is already introduced. We just changing default
See soc/riscv/riscv-privilege/virt/Kconfig.defconfig.series

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc [hunk 6]
```diff
@@ -16,6 +16,10 @@ config B9xCPU_RISCV32
 
 endchoice
 
+config TELINK_B92_PMP
+	bool "Support Hardware Memory Protection"
+	select RISCV_PMP
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/Kconfig.soc:22]: depends on SOC_RISCV_TELINK_B92
help ...

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**